step-cli/step-cli-launcher/src/main/java/step/cli/StepConsole.java

```diff
@@ -209,11 +217,34 @@ private void checkApFolder(File param) throws IOException {
                 }
             }
 
+            protected ControllerServicesClient createControllerServicesClient(){
+                return new ControllerServicesClient(getControllerCredentials());
+            }
+
+            protected ControllerCredentials getControllerCredentials() {
+                String authToken = getAuthToken();
+                return new ControllerCredentials(stepUrl, authToken == null || authToken.isEmpty() ? null : authToken);
+            }
+
+            protected void checkStepControllerVersion() {
+                try {
+                    new ControllerVersionValidator(createControllerServicesClient()).validateVersions(getVersion());
+                } catch (ControllerVersionValidator.ValidationException e) {
+                    if (!force) {
+                        log.error("To execute the command please update you CLI application or use the " + FORCE + " option to ignore this validation");
+                        throw new StepCliExecutionException("CLI version mismatch detected", e);
```

Comment:
Not sure if this message would be clear for the user. Would clearly state that the version of the controller doesn't match the version of the CLI and also print the 2 values. Proposition: 

Error: Version Mismatch
The server version (vX.X.X) is incompatible with the current CLI version (vY.Y.Y).
Please ensure both the CLI and server are running compatible versions. {{if minor mismatch => "You can use the --force option to ignore this validation"}}

Reply:
Perfect. Just added a minor remark for the warning in case of minor mismatch